### PR TITLE
Fixed not working authority matcher for kedge gRPC, Added more tags.

### PR DESCRIPTION
**Pull request by Bplotka**
Signed-off-by: Bartek Plotka <bwplotka@gmail.com>

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -4,7 +4,7 @@ go:
   - 1.8.x
   - 1.9.x
 
-go_import_path: github.com/improbable-eng/thanos
```
Comment (stefan-improbable) [inline .travis.yml]: is this ok?

File: cmd/winch/main.go [hunk 2]
```diff
@@ -138,10 +136,8 @@ func main() {
 		tlsConfig,
 		logEntry,
 		mux,
+		*flagDebugMode,
 	)
-	if *flagDebugMode {
```
Comment (stefan-improbable) [inline cmd/winch/main.go]: why is this change here

File: .travis.yml (hunk 1 above)
Comment (Bplotka) [reply .travis.yml]: Well. Not, that's why I changed this ;p

File: cmd/winch/main.go (hunk 2 above)
Comment (Bplotka) [reply cmd/winch/main.go]: to enable debug mode consistently (same for winch and grpc)

File: scripts/protogen.sh [hunk 3]
```diff
@@ -1,10 +1,10 @@
 #!/bin/bash
 # Generates protobuf Go datastructures from the proto directory.
 
-SCRIPT_DIR="$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd -P)"
-PROTOBUF_DIR=${PROTOBUF_DIR-${SCRIPT_DIR}/proto}
+ROOT_DIR="$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd -P)/.."
```
Comment (stefan-improbable) [inline scripts/protogen.sh]: This should be `+ROOT_DIR="$(cd "$(dirname "${BASH_SOURCE[0]}/..")" && pwd -P)"`

File: pkg/kedge/grpc/director/router/router.go [hunk 4]
```diff
@@ -88,15 +98,33 @@ func (r *static) serviceNameMatches(fullMethodName string, matcher string) bool
...
+func (r *static) authorityHostMatches(md metautils.NiceMD, hostMatcher string) bool {
+	if hostMatcher == "" {
+		return true
+	}
+	auth := md.Get(":authority")
+	if auth == "" {
+		return false // there was no authority header and it was expected
+	}
+
+	return stripPort(auth) == hostMatcher
```
Comment (stefan-improbable) [inline pkg/kedge/grpc/director/router/router.go]: Can you use net.SplitHostPort and also in authorityPortMatches?

File: pkg/kedge/grpc/director/router/router.go [hunk 5]
```diff
@@ -88,15 +98,33 @@ func (r *static) serviceNameMatches(fullMethodName string, matcher string) bool
...
 		return true
 	}
 	auth := md.Get(":authority")
 	if auth == "" {
 		return false // there was no authority header and it was expected
 	}
-	return auth == matcher
+
+	portUint, err := strconv.ParseUint(portOnly(auth), 10, 32)
+	if err != nil {
```
Comment (stefan-improbable) [inline pkg/kedge/grpc/director/router/router.go]: Can you log this error?

File: pkg/kedge/grpc/director/router/router.go (hunk 4 above)
Comment (Bplotka) [reply pkg/kedge/grpc/director/router/router.go]: please no (: I tried that, but it expects port to be there (and returns error)

File: pkg/kedge/grpc/director/router/router.go (hunk 5 above)
Comment (Bplotka) [reply pkg/kedge/grpc/director/router/router.go]: done

File: pkg/kedge/grpc/director/router/router.go (hunk 4 above)
Comment (stefan-improbable) [reply pkg/kedge/grpc/director/router/router.go]: What about 
```
u, e := url.Parse("null://" + auth)
if e := nil {logger.WithError(...); return false}
return u.Hostname() == hostMatcher // and similar for port
```

Comment (Bplotka) [reply pkg/kedge/grpc/director/router/router.go]: I would rather avoid these hacks. I added test case for IPv6. My functions seems to work just fine (: Is that enough?
